### PR TITLE
Add vpa support

### PR DESCRIPTION
**Pull request by huizhifan**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create --enable-vpa
az aks update --enable-vpa
az

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by huizhifan** [comment]
Kicked off a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=60442272&view=results) to generate the recording file though not sure if I did it right.

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -835,6 +835,12 @@
         - name: --private-dns-zone
           type: string
           short-summary: The private dns zone mode for private cluster.
+        - name: --enable-vpa
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_help.py:838]: Missing help for this in aks create?

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 2]
```diff
@@ -2689,6 +2762,33 @@ def update_node_restriction(self, mc: ManagedCluster) -> ManagedCluster:
...
+    def update_vpa(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update workload auto-scaler profile vertical pod auto-scaler for the ManagedCluster object.
+        
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_vpa():
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2774]:
```suggestion
                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile()
```

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 3]
```diff
@@ -2689,6 +2762,33 @@ def update_node_restriction(self, mc: ManagedCluster) -> ManagedCluster:
...
+            if mc.workload_auto_scaler_profile.vertical_pod_autoscaler is None:
+                mc.workload_auto_scaler_profile.vertical_pod_autoscaler = self.models.ManagedClusterWorkloadAutoScalerProfileVerticalPodAutoscaler()
+
+            # set enabled
+            mc.workload_auto_scaler_profile.vertical_pod_autoscaler.enabled = True
+
+        if self.context.get_disable_vpa():
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2783]:
```suggestion
                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile()
```

**Comment by FumingZhang** [comment]
> Kicked off a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=60442272&view=results) to generate the recording file though not sure if I did it right.

New [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=60455673&view=results)

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 2 above)
Comment (huizhifan) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: Thanks for the comment. Updated in the latest iteration.

File: src/aks-preview/azext_aks_preview/_help.py [hunk 4]
```diff
@@ -469,6 +469,9 @@
         - name: --defender-config
           type: string
           short-summary: Path to JSON file containing Microsoft Defender profile configurations.
+        - name: --enable-vpa
+          type: bool
+          short-summary: Enable vertical pod autoscaler for cluster.
```
Comment (xiazhan) [inline src/aks-preview/azext_aks_preview/_help.py:474]: Add --disable-vpa as well?

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 5]
```diff
@@ -2253,6 +2311,20 @@ def set_up_node_restriction(self, mc: ManagedCluster) -> ManagedCluster:
...
+        """Set up workload auto-scaler profile vpa for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_vpa():
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile()
+            mc.workload_auto_scaler_profile.vertical_pod_autoscaler = self.models.ManagedClusterWorkloadAutoScalerProfileVerticalPodAutoscaler(enable=True)
```
Comment (xiazhan) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2324]: Check if mc.workload_auto_scaler_profile.vertical_pod_autoscaler is None?

**Comment by FumingZhang** [comment]
> > Kicked off a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=60442272&view=results) to generate the recording file though not sure if I did it right.
> 
> New [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=60455673&view=results)

Another [one](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=60510497&view=results), if this passed, please commit the recording file from pipeline artifact to `src/aks-preview/azext_aks_preview/tests/latest/recordings`.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 5 above)
Comment (huizhifan) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: Updated in the latest iteration

File: src/aks-preview/HISTORY.rst [hunk 6]
```diff
@@ -14,6 +14,13 @@ Pending
 
 * Update to use 2022-08-02-preview api version.
 
++++++++
+
+0.5.101
++++++++
+
+* Add --enable-vpa/--disable-vpa to enable/disable vertical pod autoscaler feature.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:22]: Conflict? There's already 0.5.101 in [main](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/HISTORY.rst) branch. Also, if you want to release a new version, please also update `VERSION` in `setup.py`.

Comment (huizhifan) [reply src/aks-preview/HISTORY.rst]: Updated in the latest iteration

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**